pkg/apply/filter/current-uids-filter_test.go

```diff
@@ -50,19 +48,8 @@ func TestCurrentUIDFilter(t *testing.T) {
 			}
 			obj := defaultObj.DeepCopy()
 			obj.SetUID(types.UID(tc.objUID))
-			actual, reason, err := filter.Filter(obj)
-			if err != nil {
-				t.Fatalf("CurrentUIDFilter unexpected error (%s)", err)
-			}
-			if tc.filtered != actual {
-				t.Errorf("CurrentUIDFilter expected filter (%t), got (%t)", tc.filtered, actual)
-			}
-			if tc.filtered && len(reason) == 0 {
-				t.Errorf("CurrentUIDFilter filtered; expected but missing Reason")
-			}
-			if !tc.filtered && len(reason) > 0 {
-				t.Errorf("CurrentUIDFilter not filtered; received unexpected Reason: %s", reason)
-			}
+			err := filter.Filter(obj)
+			testutil.AssertEqual(t, tc.expectedError, err)
```

Comment:
Now that we're using typed errors with the `Is` method implemented, we can use a simple AssertEqual and the error comparator will handle equality comparisons, even if the pointer doesn't point to the exact same error instance or even if it's wrapped in other errors.